src/convenience.jl

```diff
@@ -17,5 +17,6 @@ _sourcetype(filenames::NamedTuple) = _sourcetype(first(filenames))
 
 # Internal read method
 function _open(f, filename::AbstractString; kw...)
-    _open(f, _sourcetype(filename), filename; kw...)
+    source=get(kw,:source,_sourcetype(filename))
```

Comment:
You can just make a keyword argument for `_open(f, filename::Abstractstring; source=_sourcetype(filename), kw...)`

Reply:
Done.